packages/@react-spectrum/listbox/src/ListBoxOption.tsx

```diff
@@ -44,7 +44,7 @@ export function ListBoxOption<T>(props: OptionProps<T>) {
     rendered,
     key
   } = item;
-
+  let domProps = filterDOMProps(item.props);
```

Comment:
I was a bit too late in reviewing this, but did you check if this means our generated id from useOption gets overridden by a user provided id? We'll want to make sure it doesn't cuz I think ComboBox relies on the ids of the options to be formatted in a specific way to generate a proper `aria-activedescendant`

Reply:
Good catch, let me push a fix. I'll make a note on the audit item to watch out for this.

Reply:
PR created here: https://github.com/adobe/react-spectrum/pull/4179